src/Helper/Database/Refactorings/ConnectionCallFactory.php

```diff
@@ -11,22 +11,19 @@
 use TYPO3\CMS\Core\Database\ConnectionPool;
 use TYPO3\CMS\Core\Utility\GeneralUtility;
 
-trait ConnectionCallTrait
+final class ConnectionCallFactory
 {
     /**
      * @var NodeFactory
      */
-    protected $nodeFactory;
+    private $nodeFactory;
 
-    /**
-     * @required
-     */
-    public function autowireNodeFactoryTrait(NodeFactory $nodeFactory): void
+    public function __construct(NodeFactory $nodeFactory)
     {
         $this->nodeFactory = $nodeFactory;
     }
```

Comment:
Traits depend on magical presence of other properties, e.g. here is "hope" for $nodeFactory.
Instead, it's better to use service, that has clear contract and there is no need for hope :)

I know this was used in Rector a lot, so it's only natural to use it in the code that build on Rector. But we're moving away from this in Rector, so code is cleaner and easier to maintain.

Reply:
@TomasVotruba Yes, i am not a big fan of traits either. So cool, Rector is moving away from it.